homeassistant/components/http/__init__.py

```diff
@@ -403,30 +431,58 @@ async def redirect(request: web.Request) -> web.StreamResponse:
             self.app.router.add_route("GET", url, redirect)
         )
 
-    def register_static_path(
-        self, url_path: str, path: str, cache_headers: bool = True
+    def _make_static_resources(
+        self, configs: Collection[StaticPathConfig]
+    ) -> dict[str, CachingStaticResource | web.StaticResource | None]:
+        """Create a list of static resources."""
+        return {
+            config.url_path: _STATIC_CLASSES[config.cache_headers](
+                config.url_path, config.path
+            )
+            if os.path.isdir(config.path)
+            else None
+            for config in configs
+        }
+
+    async def async_register_static_paths(
```

Comment:
This needs direct test coverage as it only has indirect coverage and only `register_static_path` is covered directly